arch/Kconfig

```diff
@@ -230,57 +230,7 @@ endmenu # Custom Chip Configuration
 endif #ARCH_CHIP_CUSTOM
 
 source "arch/dummy/Kconfig"
-
-config ARCH_TOOLCHAIN_IAR
-	bool
-	default n
-
-config ARCH_TOOLCHAIN_GNU
-	bool
-	default n
-
-config ARCH_TOOLCHAIN_CLANG
-	bool
-	select ARCH_TOOLCHAIN_GNU
-	default n
-
-choice
-	prompt "Link Time Optimization (LTO)"
-	default LTO_NONE
-	---help---
-		This option enables Link Time Optimization (LTO), which allows the
-		compiler to optimize binaries globally.
-
-		If unsure, select LTO_NONE. Note that LTO is very resource-intensive
-		so it's disabled by default.
-
-config LTO_NONE
-	bool "None"
-	---help---
-		Build the kernel normally, without Link Time Optimization (LTO).
-
-config LTO_FULL
-	bool "GNU Full LTO (EXPERIMENTAL)"
-	depends on ARCH_TOOLCHAIN_GNU
-	---help---
-		Link time optimization is implemented as a GCC front end for a bytecode
-		bytecode representation of GIMPLE that is emitted in special sections
-		of .o files. Currently, LTO support is enabled in most ELF-based systems,
-		as well as darwin, cygwin and mingw systems.
-
-config LTO_THIN
-	bool "Clang ThinLTO (EXPERIMENTAL)"
-	depends on ARCH_TOOLCHAIN_CLANG
-	---help---
-		This option enables Clang's ThinLTO, which allows for parallel
-		optimization and faster incremental compiles compared to the
-		CONFIG_LTO_FULL option. More information can be found
-		from Clang's documentation:
-
-		https://clang.llvm.org/docs/ThinLTO.html
-
-		If unsure, say Y.
-endchoice
+source "arch/Kconfig.toolchain"
```

Comment:
why split to Kconfig.toolchain?

Reply:
Just separates the toolchain related configuration into Kconfig.toolchain, new toolchain support can be added to this file